diff --git a/src/rcore.c b/src/rcore.c
index 58c8c4ce..d948fc90 100644
--- a/src/rcore.c
+++ b/src/rcore.c
@@ -3469,16 +3469,23 @@ bool IsKeyPressed(int key)
 {
     bool pressed = false;
 
+#if defined(PLATFORM_ANDROID)
+    pressed = (GetKeyPressed() == key);
+#else
     if ((CORE.Input.Keyboard.previousKeyState[key] == 0) && (CORE.Input.Keyboard.currentKeyState[key] == 1)) pressed = true;
-
+#endif
     return pressed;
 }
 
 // Check if a key is being pressed (key held down)
 bool IsKeyDown(int key)
 {
+#if defined(PLATFORM_ANDROID)
+    return IsKeyPressed(key);
+#else
     if (CORE.Input.Keyboard.currentKeyState[key] == 1) return true;
     else return false;
+#endif
 }
 
 // Check if a key has been released once
@@ -3714,7 +3721,7 @@ bool IsMouseButtonUp(int button)
 int GetMouseX(void)
 {
 #if defined(PLATFORM_ANDROID)
-    return (int)CORE.Input.Touch.position[0].x;
+    return (int)(CORE.Input.Touch.position[0].x*CORE.Window.screen.width - CORE.Window.renderOffset.x/2);
 #else
     return (int)((CORE.Input.Mouse.currentPosition.x + CORE.Input.Mouse.offset.x)*CORE.Input.Mouse.scale.x);
 #endif
@@ -3724,7 +3731,7 @@ int GetMouseX(void)
 int GetMouseY(void)
 {
 #if defined(PLATFORM_ANDROID)
-    return (int)CORE.Input.Touch.position[0].y;
+    return (int)(CORE.Input.Touch.position[0].y*CORE.Window.screen.height - CORE.Window.renderOffset.y/2);
 #else
     return (int)((CORE.Input.Mouse.currentPosition.y + CORE.Input.Mouse.offset.y)*CORE.Input.Mouse.scale.y);
 #endif
@@ -3846,13 +3853,13 @@ Vector2 GetTouchPosition(int index)
 
     if ((CORE.Window.screen.width > CORE.Window.display.width) || (CORE.Window.screen.height > CORE.Window.display.height))
     {
-        position.x = position.x*((float)CORE.Window.screen.width/(float)(CORE.Window.display.width - CORE.Window.renderOffset.x)) - CORE.Window.renderOffset.x/2;
-        position.y = position.y*((float)CORE.Window.screen.height/(float)(CORE.Window.display.height - CORE.Window.renderOffset.y)) - CORE.Window.renderOffset.y/2;
+        position.x = position.x*(float)CORE.Window.screen.width - CORE.Window.renderOffset.x/2;
+        position.y = position.y*(float)CORE.Window.screen.height - CORE.Window.renderOffset.y/2;
     }
     else
     {
-        position.x = position.x*((float)CORE.Window.render.width/(float)CORE.Window.display.width) - CORE.Window.renderOffset.x/2;
-        position.y = position.y*((float)CORE.Window.render.height/(float)CORE.Window.display.height) - CORE.Window.renderOffset.y/2;
+        position.x = position.x*(float)CORE.Window.render.width - CORE.Window.renderOffset.x/2;
+        position.y = position.y*(float)CORE.Window.render.height - CORE.Window.renderOffset.y/2;
     }
 #endif
 #if defined(PLATFORM_WEB) || defined(PLATFORM_RPI) || defined(PLATFORM_DRM)
